src/main/java/htsjdk/samtools/seekablestream/SeekablePathStream.java

```diff
@@ -0,0 +1,108 @@
+package htsjdk.samtools.seekablestream;
+
+import htsjdk.samtools.util.Log;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class SeekablePathStream extends SeekableStream {
```

Comment:
Add direct unit tests for this class that fully exercise `read()` (both versions), `skip()`, `seek()`, `position()`, and `length()` (can use a `Path` backed by a regular file for the tests).